keyvi/tests/keyvi/dictionary/fuzzy_match_test.cpp

```diff
@@ -0,0 +1,240 @@
+/* * keyvi - A key value store.
+ *
+ * Copyright 2018   Narek Gharibyan<narekgharibyan@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * fuzzy_match_test.cpp
+ *
+ *  Created on: February 11, 2018
+ *      Author: Narek Gharibyan <narekgharibyan@gmail.com>
+ */
+
+#include <boost/test/unit_test.hpp>
+
+#include "dictionary/dictionary.h"
+#include "testing/temp_dictionary.h"
+
+namespace keyvi {
+namespace dictionary {
+
+BOOST_AUTO_TEST_SUITE(FuzzyMatchTests)
```

Comment:
could you add some test cases for edge cases:

- no match
- empty string
- query shorter than prefix length

some CJK would be nice as well, although you use umlauts.

Reply:
some tests for the score would be nice as well. E.g. when it comes to non-ascii it is important that e.g. a mismatch of 'ü' scores edit distance 1 (codepoint length) and not 2 (byte length).

Reply:
Sure!